### PR TITLE
Execute block on receiver together with exclusive mode

### PR DESCRIPTION
**Pull request by Burgestrand**
This has been rebased on 0.17.5-release. See #604 for previous discussion.

### CODE REVIEW

**Comment by digitalextremist** [comment]
@Burgestrand thank you very much for doing this!

**Comment by ioquatix** [comment]
Can we merge this?

File: lib/celluloid/calls.rb [hunk 1]
```diff
@@ -23,9 +15,13 @@ def execute_block_on_receiver
     end
 
     def dispatch(obj)
-      check(obj)
-      _b = @block && @block.to_proc
-      obj.public_send(@method, *@arguments, &_b)
+      begin
+        check(obj)
+      rescue => error
+        raise AbortError, error
```
Comment (tarcieri) [inline lib/celluloid/calls.rb]: This seems broken to me. It means that the `ArgumentErrors` below will only crash the receiver (but not the caller?)

Comment (ioquatix) [inline lib/celluloid/calls.rb]: Could you add a test to cover this behaviour so we can rebase and see it break?

Comment (Burgestrand) [inline lib/celluloid/calls.rb]: Yes, will do. This does indeed seem broken.

Comment (Burgestrand) [inline lib/celluloid/calls.rb]: From what I can see this is already being tested here: https://github.com/elabs/celluloid/blob/0.17.5-release_allow-receiver-blocks-in-exclusive-mode/spec/celluloid/calls_spec.rb — ArgumentError and NoMethodErrors properly crashing the sender and not the receiver. If I recall correctly (it's been a few months, memory rusty), the AbortError is specifically to crash the sender and not the receiver. Catching the errors when doing check, as opposed to inside the check-method, should make no difference.

Furthermore I tried rebasing this on 0.18.0-prerelease, but I had some conflicts so I stopped that endeavour.

Finally, I'm sorry, but I believe I'll personally be dropping this issue. Thanks for the work you've put into all of this, but at least for the coming few weeks/months I'm done.

**Comment by ioquatix** [comment]
Shouldn't we merge this?

**Comment by tarcieri** [comment]
It needs a rebase to be mergeable. I can push it back up and reopen the PR.

**Comment by ioquatix** [comment]
Ah yep, perhaps I will have time next week I will take a look if nothing has happened by then.

**Pull request closed without merging**